shark_turbine/kernel/wave/codegen.py

```diff
@@ -495,7 +539,10 @@ def handle_write(emitter: WaveEmitter, node: fx.Node):
             is_read=False,
         )
 
-        vector_d.scatter(kb_dest, start_indices, offsets_vec, mask, insert_vector)
+        if elements_per_thread == 1:
```

Comment:
Does this mean that we cannot codegen contiguous writes of more than 1 element? If so, seems very restrictive.

Reply:
Contiguous writes are covered in separate branch above, added this case because `partition_strided_operators` tends to generate a bunch of single-element writes and we don't need `scatter` for 1-element writes a 0th scatter index is always 0.

Reply:
makes sense, thanks for clarifying.